ateam_kenobi/src/play_helpers/robot_assignment.hpp

```diff
@@ -0,0 +1,39 @@
+// Copyright 2024 A Team
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+
+#ifndef PLAY_HELPERS__ROBOT_ASSIGNMENT_HPP_
+#define PLAY_HELPERS__ROBOT_ASSIGNMENT_HPP_
+
+#include <vector>
+#include <optional>
+#include "types/robot.hpp"
+
+namespace ateam_kenobi::play_helpers
+{
+
+std::vector<std::optional<Robot>> assignRobots(
+  const std::vector<Robot> & available_robots,
+  const std::vector<ateam_geometry::Point> & positions,
+  const std::vector<std::vector<int>> & disallowed_robot_ids = {});
```

Comment:
```suggestion
/*
  Greedily assign robots to available positions
*/
std::vector<std::optional<Robot>> assignRobots(
  const std::vector<Robot> & available_robots,
  const std::vector<ateam_geometry::Point> & positions,
  const std::vector<std::vector<int>> & disallowed_robot_ids = {});
```

Reply:
The greedy part is more of an implementation detail that will go away when I can rewrite this to use your new assignment algorithm.

Reply:
Sounds good